### PR TITLE
Add integration test for empty result

### PR DESCRIPTION
**Pull request by mukeshk**
Addressed to #1066

### CODE REVIEW

File: src/test/robotframework/series/search/logic.robot [hunk 1]
```diff
@@ -0,0 +1,19 @@
+*** Settings ***
+Documentation    Searching Series by Catalog Number and Catalog Name
```
Comment (php-coder) [inline src/test/robotframework/series/search/logic.robot]: Could you, please, rephrase it to "Verify series search scenarios"?

File: src/test/robotframework/series/search/logic.robot [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+Library          SeleniumLibrary
+Resource         ../../selenium.utils.robot
+Suite Setup      Before Test Suite
+Suite Teardown   Close Browser
+Force Tags       series  search  validation
```
Comment (php-coder) [inline src/test/robotframework/series/search/logic.robot]: Please, replace "validation" tag to "logic"

File: src/test/robotframework/series/search/logic.robot [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+*** Test Cases ***
+Search non existing catalog number results in no series found
+	Input Text              id=catalogNumber  888
+	Select Random Option From List    catalogName
```
Comment (php-coder) [inline src/test/robotframework/series/search/logic.robot]: Only for this line: please, use 2 spaces (instead of 4) to separate a keyword and it's parameter.

File: src/test/robotframework/series/search/logic.robot [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+	Submit Form             id=search-series-form
+	Element Text Should Be  id=no-series-found  No series found
```
Comment (php-coder) [inline src/test/robotframework/series/search/logic.robot]: To make the test cases more readable, could you start all the parameters from the position of the first parameter of the longest keyword? Like we have in "Before Test Suite" keyword below.

File: src/test/robotframework/series/search/logic.robot [hunk 5]
```diff
@@ -0,0 +1,19 @@
+*** Settings ***
+Documentation    Searching Series by Catalog Number and Catalog Name
+Library          SeleniumLibrary
+Resource         ../../selenium.utils.robot
+Suite Setup      Before Test Suite
+Suite Teardown   Close Browser
```
Comment (php-coder) [inline src/test/robotframework/series/search/logic.robot]: There 3 spaces between keywords and its parameter, please, reduce it to 2 spaces and align other parameters in the Settings section in the same way. Like I did in the bd80c86d34ea9d commit.

File: src/test/robotframework/series/search/logic.robot [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+Force Tags       series  search  validation
+
+*** Test Cases ***
+Search non existing catalog number results in no series found
```
Comment (php-coder) [inline src/test/robotframework/series/search/logic.robot]: Let's change the test case name a little bit: "Search series by non-existing catalog number"

File: src/test/robotframework/selenium.utils.robot [hunk 7]
```diff
@@ -66,3 +66,11 @@ Wait Until Element Value Is
 	...                 ${text}
 	...                 ';
 	Wait For Condition  ${elemHasValue}
+
+Select Random Option From List
+	[Documentation]     Select any random item from the list.
```
Comment (php-coder) [inline src/test/robotframework/selenium.utils.robot]: "Select any random item from the list." -> "Choose a random option from a select element"

File: src/test/robotframework/selenium.utils.robot [hunk 8]
```diff
@@ -66,3 +66,11 @@ Wait Until Element Value Is
...
+    [Arguments]         ${listName}    
+	${listItems}=  		Get List Items    ${listName}
+    ${listSize}=  		Get Length    ${listItems}
+    ${listItemIndex}=	Evaluate  random.randint(0,${listSize}-1)    random
+	Select from list by index   ${listName}    ${listItemIndex}
```
Comment (php-coder) [inline src/test/robotframework/selenium.utils.robot]: 1) "Select from list by index" -> "Select from list by index"
2) please, use 2 spaces to separate parameters

File: src/test/robotframework/selenium.utils.robot [hunk 9]
```diff
@@ -66,3 +66,11 @@ Wait Until Element Value Is
 	...                 ${text}
 	...                 ';
 	Wait For Condition  ${elemHasValue}
+
+Select Random Option From List
+	[Documentation]     Select any random item from the list.
+    [Arguments]         ${listName}    
+	${listItems}=  		Get List Items    ${listName}
+    ${listSize}=  		Get Length    ${listItems}
+    ${listItemIndex}=	Evaluate  random.randint(0,${listSize}-1)    random
```
Comment (php-coder) [inline src/test/robotframework/selenium.utils.robot]: 1. use 2 spaces to separate parameters
2. let's add a space between randint() arguments: `random.randint(0, ${listSize}-1)`
3. let's explicitly name the last argument to give a hint what it is used for: `modules=random` (I hope it should work that way)
4. let's rename `listItemIndex` to `randomIndex`

File: src/test/robotframework/selenium.utils.robot [hunk 10]
```diff
@@ -66,3 +66,11 @@ Wait Until Element Value Is
 	...                 ${text}
 	...                 ';
 	Wait For Condition  ${elemHasValue}
+
+Select Random Option From List
+	[Documentation]     Select any random item from the list.
+    [Arguments]         ${listName}
```
Comment (php-coder) [inline src/test/robotframework/selenium.utils.robot]: Please, rename `${listName}` argument to `${locator}` to express its meaning and possible values.

File: src/test/robotframework/selenium.utils.robot [hunk 11]
```diff
@@ -66,3 +66,11 @@ Wait Until Element Value Is
 	...                 ${text}
 	...                 ';
 	Wait For Condition  ${elemHasValue}
+
+Select Random Option From List
```
Comment (php-coder) [inline src/test/robotframework/selenium.utils.robot]: Please, align parameters in this keyword after you finish with my comments below.

File: src/test/robotframework/selenium.utils.robot [hunk 12]
```diff
@@ -66,3 +66,11 @@ Wait Until Element Value Is
...
+	[Documentation]     Select any random item from the list.
+    [Arguments]         ${listName}    
+	${listItems}=  		Get List Items    ${listName}
```
Comment (php-coder) [inline src/test/robotframework/selenium.utils.robot]: Let's make it simple: rename `listItems` to `options`

File: src/test/robotframework/selenium.utils.robot [hunk 13]
```diff
@@ -66,3 +66,11 @@ Wait Until Element Value Is
...
+    ${listSize}=  		Get Length    ${listItems}
```
Comment (php-coder) [inline src/test/robotframework/selenium.utils.robot]: listSize -> size

**Comment by php-coder** [comment]
@mukeshk here is what bot says:

> danger check: pull request description doesn’t contain a link to original issue.

Note that it says about "description" and not "title" ;)

**Comment by php-coder** [comment]
P.S. please, edit also the commit's message: instead of "#1066" use "Fix #1066", so the related issue will be closed automatically when this commit will be merged.

**Comment by php-coder** [comment]
@mukeshk Thank you! I'll review it tonight.

**Comment by mukeshk** [comment]
Not sure about the commit message
Title : test(series/search/logic.robot):Add Integration test for empty result 
Desc: Address to 1066

Should I change the description
Address to #1066 and Fix#1066 or is it good at this point of time.

**Comment by php-coder** [comment]
@mukeshk the bot complain about PR description (not about commit description).

So, in the commit message you should use "Fix #XXX" but in the PR message "Addressed to #XXX".

File: src/test/robotframework/selenium.utils.robot [hunk 14]
```diff
@@ -66,3 +66,11 @@ Wait Until Element Value Is
 	...                 ${text}
 	...                 ';
 	Wait For Condition  ${elemHasValue}
+
+Choose a random option from a select element
```
Comment (php-coder) [inline src/test/robotframework/selenium.utils.robot]: Please, rename it back to "Select Random Option From List"

File: src/test/robotframework/selenium.utils.robot [hunk 15]
```diff
@@ -66,3 +66,11 @@ Wait Until Element Value Is
...
+	[Documentation]            Selects a random option from a select element
```
Comment (php-coder) [inline src/test/robotframework/selenium.utils.robot]: You've renamed the keyword but I asked for documentation line:
"Selects a random option from a select element" -> "Choose a random option from a select element"

File: src/test/robotframework/series/search/logic.robot [hunk 16]
```diff
@@ -0,0 +1,19 @@
...
+Library         SeleniumLibrary
+Resource        ../../selenium.utils.robot
+Suite Setup     Before Test Suite
+Suite Teardown  Close Browser
+Force Tags      series  search  logic
+
+*** Test Cases ***
+Search series by non-existing catalog number
+	Input Text                                    id=catalogNumber  888
+	Choose a random option from a select element  catalogName
```
Comment (php-coder) [inline src/test/robotframework/series/search/logic.robot]: Because the argument is element location, let's explicitly declare that we want to search an element by id:
catalogName -> id=catalogName

File: src/test/robotframework/selenium.utils.robot (hunk 8 above)
Comment (php-coder) [reply src/test/robotframework/selenium.utils.robot]: > "Select from list by index" -> "Select from list by index"

^^^ this one is still requires your attention :)

**Comment by php-coder** [comment]
Could you also, please, modify the commit message so I'll be able to merge your changes easily.]

Instead of:
```
test(series/search/logic.robot):Add Integration test for empty result

Address to 1066
```

it should be
```
test(series/search/logic.robot): add test for the empty search results.

Fix #1066
```

**Comment by php-coder** [comment]
P.S. please, ping me here when the PR will be ready for next round of review :)

File: src/test/robotframework/selenium.utils.robot (hunk 8 above)
Comment (mukeshk) [reply src/test/robotframework/selenium.utils.robot]: <3spaces>Select from list by index<2spaces>${locator}<2spaces>${randomIndex}

Not sure if I am missing something here.

Comment (php-coder) [reply src/test/robotframework/selenium.utils.robot]: Ah, it's my mistake. I meant

"Select from list by index" -> "Select From List By Index"

Capitalize all the words.

I'm sorry, it was copy&paste&forget bug :)

**Comment by mukeshk** [comment]
Aha ok. Sure no problem.

**Comment by php-coder** [comment]
@mukeshk and could you also edit commit message a little?

I'm not sure that this pattern would work: `Fix#1066`
Should be `Fix #1066`

**Comment by mukeshk** [comment]
Sure, will do it

**Comment by mukeshk** [comment]
Implemented the suggestions.

**Comment by php-coder** [comment]
>> @mukeshk and could you also edit commit message a little?
>
>I'm not sure that this pattern would work: Fix#1066
> Should be Fix #1066

How about amending commit message to add a missing space? ;)

**Comment by mukeshk** [comment]
image

Is this good. Last time I tried it did not reflect. Can you check if this is good?
